Yoast/yoast-components
# Updated React-Dom dependency to latest version

## Summary

This PR can be summarized in the following changelog entry:

* N / A. Updated React and React-dom to the same version as WordPress (16.6.3). 

## Relevant technical choices:

* Also updated "enzyme-adapter-react-16" to its next minor version to fix failing tests. 

## Test instructions
This PR can be tested by following these steps:

* Nothing should have changed so there is no real scenario to be tested. A simple test to check nothing has broken should suffice. 
* Build all the files. If you want to be super sure, remove the `node_modules` folder first. 
* All the tests should still pass, no functionality should have changed and there should be no console errors when using the plugin. 

## UI changes

Fixes #841
